### PR TITLE
[API] Add Neural Network Model Construction with Configuration File

### PR DESCRIPTION
**Pull request by jijoongmoon**
- Add Construction of Nerual Network Model with Configuration File.
 . ml_nnmodel_construct_with_conf()
 . Add one positive and one negative tests

**Self evaluation:**
1. Build test:	 [X]Passed [ ]Failed [ ]Skipped
2. Run test:	 [X]Passed [ ]Failed [ ]Skipped

Signed-off-by: jijoong.moon <jijoong.moon@samsung.com>

### CODE REVIEW

File: api/capi/src/nntrainer.cpp [hunk 1]
```diff
@@ -77,6 +77,30 @@ int ml_nnmodel_construct(ml_nnmodel_h *model) {
...
+int ml_nnmodel_construct_with_conf(const char *model_conf, ml_nnmodel_h *model) {
+  int status = ML_ERROR_NONE;
+  ml_nnmodel *nnmodel;
+
+  std::ifstream conf_file(model_conf);
+  if (!conf_file.good()) {
+    std::cerr << "Error: Cannot open model configuration file : " << model_conf << std::endl;
+    return ML_ERROR_INVALID_PARAMETER;
+  }
+  if (!model) {
```
Comment (zhoonit) [inline api/capi/src/nntrainer.cpp:89]: I think this should be `if(model)`

File: test/tizen_capi/unittest_tizen_capi.cpp [hunk 2]
```diff
@@ -46,6 +46,32 @@ TEST(nntrainer_nnmodel_construct_deconstruct, nntrainer_02_n) {
   EXPECT_EQ(status, ML_ERROR_INVALID_PARAMETER);
 }
 
+/**
+ * @brief Neural Network Model Construct wit Configuration File Test
+ * @return 0 success, -EINVAL if failed.
+ */
+TEST(nntrainer_nnmodel_construct_deconstruct, nntrainer_04_n) {
+  ml_nnmodel_h handle;
+  const char *model_conf = "/test/test.ini";
```
Comment (zhoonit) [inline test/tizen_capi/unittest_tizen_capi.cpp:55]: I think it is good to clarify this test is negative test by changing `/test/test.ini` -> `/test/wrong_conf.ini`.

File: api/capi/include/nntrainer.h [hunk 3]
```diff
@@ -70,6 +70,19 @@ typedef enum {
  */
 int ml_nnmodel_construct(ml_nnmodel_h *model);
 
+/**
+ * @brief Constructs the neural network model with configuration file.
+ * @details Use this function to create Neural Netowrk Model.
+ * @since_tizen 5.5
```
Comment (helloahn) [inline api/capi/include/nntrainer.h:76]: will tizen 5.5 contains NNtrainer APIs?

File: api/capi/src/nntrainer.cpp (hunk 1 above)
Comment (jijoongmoon) [reply api/capi/src/nntrainer.cpp:89]: My mistake. This should be removed. Thanks.

File: test/tizen_capi/unittest_tizen_capi.cpp (hunk 2 above)
Comment (jijoongmoon) [reply test/tizen_capi/unittest_tizen_capi.cpp:55]: That's good. I will apply.

File: api/capi/include/nntrainer.h (hunk 3 above)
Comment (jijoongmoon) [reply api/capi/include/nntrainer.h:76]: Thanks.

**Pull request merged**